### PR TITLE
fix HIP-clang compile

### PR DESCRIPTION
**Pull request by psychocoderHPC**
HIP-clang requires an explicit implemented destructor for the random
number generator.

### CODE REVIEW

File: include/alpaka/rand/RandUniformCudaHipRand.hpp [hunk 1]
```diff
@@ -77,6 +77,10 @@ namespace alpaka
                     {
                     }
 
+#if BOOST_COMP_HIP
+                    //-----------------------------------------------------------------------------
+                    ALPAKA_FN_HOST_ACC ~Xor(){}
```
Comment (sbastrakov) [inline include/alpaka/rand/RandUniformCudaHipRand.hpp]: `= default` does not work?

Comment (psychocoderHPC) [reply include/alpaka/rand/RandUniformCudaHipRand.hpp]: I changed to `= default`

**Review by BenjaminW3 — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
@BenjaminW3 where is the test linux_clang-6_cuda-9.0_debug_analysis coming from? The test is not marked as CI or is this a gihub action job.

**Comment by BenjaminW3** [comment]
It was part of the github actions CI.
It has been removed but is still marked as required. I will fix this.

**Pull request merged**